src/main/scripts/prepare-release.py

```diff
@@ -0,0 +1,143 @@
+
+import git
+import logging as log
+import os
+import re
+import subprocess
+import sys
+from argparse import ArgumentParser
+from datetime import datetime
+from ruamel.yaml import YAML
+from tempfile import mkstemp
+
+products = ["bamboo", "bamboo-agent",
+            "bitbucket", "confluence", "crowd", "jira"]
+prodbase = "src/main/charts"
+
+
+# TODO: As a first pass this currently only filters anything tagged
+# with 'CLIP-nnn'. However there are other options; see
+# https://hello.atlassian.net/wiki/spaces/DCD/pages/2108707663/DACI+Automating+the+Helm+release+process
+def changelog_filter(log_entry):
+    return re.match(r'^\*\s+CLIP-[0-9]+', log_entry) != None
+
+
+def gen_changelog(repo_path):
+    repo = git.Repo(repo_path)
+    cli = git.Git(repo_path)
+
+    lasttag = repo.tags[-1]
+    tagver = re.sub(r'^[^-]+-', '', lasttag.name)
+    log.info(f'Generating changelog since {tagver}')
+
+    changelog = cli.log(f'{lasttag}..main', graph=True, pretty='format:%s', abbrev_commit=True, date='relative')
+    changelog = changelog.split('\n')
+
+    return list(filter(changelog_filter, changelog))
+
+
+def format_changelog_yaml(changelog):
+    # The artifacthub annotations are a single string, but formatted
+    # like YAML. Replacing the leading '*' with '-' should be
+    # sufficient.
+    c2 = map(lambda c: re.sub(r'^\* ', '- ', c), changelog)
+    return '\n'.join(c2)
+
+
+def update_changelog_file(version, changelog):
+    for prod in products:
+        log.info(f"Updating {prod} Changelog.md to {version}")
+
+        proddir = f"{prodbase}/{prod}"
+        clfile = f"{proddir}/Changelog.md"
+        (tmpfd, tmpname) = mkstemp(dir=proddir, text=True)
+
+        foundfirst = False
+        with open(clfile, 'r') as clfd:
+            for line in clfd:
+                if not foundfirst and re.match(r'^## [0-9]\.[0-9]\.[0-9]', line) != None:
+                    # First version line, inject ours before it
+                    foundfirst = True
+                    now = datetime.now()
+                    os.write(tmpfd, ("## %s\n\n" % version).encode())
+                    os.write(tmpfd, ("**Release date:** %s-%s-%s\n\n" % (now.year, now.month, now.day)).encode())
+                    os.write(tmpfd, '![AppVersion: 9.0.0](https://img.shields.io/static/v1?label=AppVersion&message=9.0.0&color=success&logo=)\n'.encode())
```

Comment:
Can we grab actual values from Chart.yaml instead? Both appVersion and Kubernetes version.